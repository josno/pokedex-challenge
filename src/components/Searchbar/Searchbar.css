.search-input {
	border: 1px solid black;
	height: 100%;
	padding-left: 10px;
	width: 180px;
}

.search-bar-container {
	background-color: gray;
	border-radius: 5px;
	display: flex;
	height: 50px;
	justify-content: space-evenly;
	margin-bottom: 15px;
	padding: 8px;
	width: 300px;
}

.pokemon-autocomplete-list {
	background-color: white;
	border-top: none;
	border-right: 1px solid black;
	border-bottom: 1px solid black;
	border-left: 1px solid black;
	bottom: 0px;
	height: 100px;
	list-style: none;
	padding-left: 10px;
	position: relative;
	margin: 0px;
	overflow-y: scroll;
	text-align: left;
	width: 180px;
	z-index: 999;
}

.autocomplete-choice:hover {
	background-color: #fbe250;
}
