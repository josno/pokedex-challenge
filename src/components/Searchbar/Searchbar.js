import React, { useState } from "react";

const Searchbar = (props) => {
	const [pokemon, setPokemon] = useState("");
	return (
		<div className='search-bar-wrapper'>
			<div className='search-bar-container'>
				<input
					className='search-input'
					value={pokemon}
					onChange={(e) => setPokemon(e.target.value)}
				/>
				<button onClick={() => props.onSearch(pokemon)}>Search</button>
			</div>
		</div>
	);
};

export default Searchbar;
