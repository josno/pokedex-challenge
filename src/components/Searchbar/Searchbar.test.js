import React from "react";
import ReactDOM from "react-dom";
import Searchbar from "./Searchbar";

import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("Searchbar render", () => {
	it(`renders without crashing`, () => {
		const div = document.createElement("div");
		ReactDOM.render(<Searchbar />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it(`renders the UI as expected and matches the snapshot`, () => {
		const wrapper = shallow(<Searchbar />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it(`renders as empty input field with placeholder text`, () => {
		const wrapper = mount(<Searchbar />);
		expect(wrapper.find("input").props().placeholder).toBe("Search database");
		expect(wrapper.find("input").props().value).toBeFalsy(); //Set as undefined
	});
});

describe("Searchbar pokemon list", () => {
	const fakeNameListProps = ["bulbasaur", "ivysaur", "venusaur"];
	const mockClick = jest.fn();
	const fakeFunction = jest.fn();

	it("renders a dropdown menu and list of Pokemon when input is clicked`", () => {
		const wrapper = mount(
			<Searchbar
				handleWarningMessage={fakeFunction}
				lick={mockClick}
				pokemonNameList={fakeNameListProps}
			/>
		);
		wrapper.find("input").simulate("click");
		expect(wrapper.find("ul")).toBeTruthy();
		expect(wrapper.find("bulbasaur")).toBeTruthy();
		expect(wrapper.find("venusaur")).toBeTruthy();
		expect(wrapper.find("ivysaur")).toBeTruthy();
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("removes dropdown menu on click of cancel button", () => {
		const wrapper = mount(
			<Searchbar
				handleWarningMessage={fakeFunction}
				onClick={mockClick}
				pokemonNameList={fakeNameListProps}
			/>
		);
		wrapper.find("input").simulate("click");
		wrapper.find("button").at(0).simulate("click");
		expect(wrapper.find("ul").exists()).toBeFalsy(); //Pokemon list has been removed
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
