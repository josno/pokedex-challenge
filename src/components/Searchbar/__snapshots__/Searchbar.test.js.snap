// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar pokemon list removes dropdown menu on click of cancel button 1`] = `
<Searchbar
  onClick={[MockFunction]}
  pokemonNameList={
    Array [
      "bulbasaur",
      "ivysaur",
      "venusaur",
    ]
  }
>
  <section
    className="search-bar-container"
  >
    <div
      className="search-input-container"
    >
      <input
        className="search-input"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="Search database"
        value=""
      />
    </div>
    <Buttons
      buttonClassName="search-buttons"
      handleClick={[Function]}
      label={
        <AiOutlineClose
          className="search-button-style"
        />
      }
    >
      <button
        className="search-buttons"
        onClick={[Function]}
      >
        <AiOutlineClose
          className="search-button-style"
        >
          <IconBase
            attr={
              Object {
                "viewBox": "0 0 1024 1024",
              }
            }
            className="search-button-style"
          >
            <svg
              className="search-button-style"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                key="0"
              />
            </svg>
          </IconBase>
        </AiOutlineClose>
      </button>
    </Buttons>
    <Buttons
      buttonClassName="search-buttons"
      handleClick={[Function]}
      label={
        <AiOutlineSearch
          className="search-button-style"
        />
      }
    >
      <button
        className="search-buttons"
        onClick={[Function]}
      >
        <AiOutlineSearch
          className="search-button-style"
        >
          <IconBase
            attr={
              Object {
                "viewBox": "0 0 1024 1024",
              }
            }
            className="search-button-style"
          >
            <svg
              className="search-button-style"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                key="0"
              />
            </svg>
          </IconBase>
        </AiOutlineSearch>
      </button>
    </Buttons>
  </section>
</Searchbar>
`;

exports[`Searchbar pokemon list renders a dropdown menu and list of Pokemon when input is clicked\` 1`] = `
<Searchbar
  onClick={[MockFunction]}
  pokemonNameList={
    Array [
      "bulbasaur",
      "ivysaur",
      "venusaur",
    ]
  }
>
  <section
    className="search-bar-container"
  >
    <div
      className="search-input-container"
    >
      <input
        className="search-input"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="Search database"
        value=""
      />
      <ul
        className="pokemon-autocomplete-list"
      >
        <li
          className="autocomplete-choice"
          key="0"
          onClick={[Function]}
          tabIndex="0"
        >
          bulbasaur
        </li>
        <li
          className="autocomplete-choice"
          key="1"
          onClick={[Function]}
          tabIndex="0"
        >
          ivysaur
        </li>
        <li
          className="autocomplete-choice"
          key="2"
          onClick={[Function]}
          tabIndex="0"
        >
          venusaur
        </li>
      </ul>
    </div>
    <Buttons
      buttonClassName="search-buttons"
      handleClick={[Function]}
      label={
        <AiOutlineClose
          className="search-button-style"
        />
      }
    >
      <button
        className="search-buttons"
        onClick={[Function]}
      >
        <AiOutlineClose
          className="search-button-style"
        >
          <IconBase
            attr={
              Object {
                "viewBox": "0 0 1024 1024",
              }
            }
            className="search-button-style"
          >
            <svg
              className="search-button-style"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                key="0"
              />
            </svg>
          </IconBase>
        </AiOutlineClose>
      </button>
    </Buttons>
    <Buttons
      buttonClassName="search-buttons"
      handleClick={[Function]}
      label={
        <AiOutlineSearch
          className="search-button-style"
        />
      }
    >
      <button
        className="search-buttons"
        onClick={[Function]}
      >
        <AiOutlineSearch
          className="search-button-style"
        >
          <IconBase
            attr={
              Object {
                "viewBox": "0 0 1024 1024",
              }
            }
            className="search-button-style"
          >
            <svg
              className="search-button-style"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                key="0"
              />
            </svg>
          </IconBase>
        </AiOutlineSearch>
      </button>
    </Buttons>
  </section>
</Searchbar>
`;

exports[`Searchbar render renders the UI as expected and matches the snapshot 1`] = `
<Fragment>
  <section
    className="search-bar-container"
  >
    <div
      className="search-input-container"
    >
      <input
        className="search-input"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="Search database"
        value=""
      />
    </div>
    <Buttons
      buttonClassName="search-buttons"
      handleClick={[Function]}
      label={
        <AiOutlineClose
          className="search-button-style"
        />
      }
    />
    <Buttons
      buttonClassName="search-buttons"
      handleClick={[Function]}
      label={
        <AiOutlineSearch
          className="search-button-style"
        />
      }
    />
  </section>
</Fragment>
`;
